dask/dataframe/core.py

```diff
@@ -5,6 +5,7 @@
 from numbers import Integral, Number
 from operator import getitem
 from pprint import pformat
+from typing import Any, Callable, Set
```

Comment:
With a `from __future__ import annotations`, you should just be able to use the builtin `set`